tests/ballerina-compiler-plugin-test/src/test/java/io/ballerina/test/compiler/plugins/TestCompilerPlugin.java

```diff
@@ -77,8 +78,9 @@ public void process(FunctionNode functionNode, List<AnnotationAttachmentNode> an
     }
 
     @Override
-    public void process(SimpleVariableNode variableNode, List<AnnotationAttachmentNode> annotations) {
-        addEvent(TestEvent.Kind.VARIAVLE_ANN, variableNode.getName().getValue(), annotations.size());
+    public void process(VariableNode variableNode, List<AnnotationAttachmentNode> annotations) {
+        addEvent(TestEvent.Kind.VARIAVLE_ANN, ((SimpleVariableNode) variableNode).getName().getValue(),
```

Comment:
This cast could also lead to issues, right?

Reply:
all the variables will desugar into simple variables, hence this should not fail

Reply:
Compiler plugins are invoked before the desugar though.

Reply:
Will have a look

Reply:
Ignored compiler plugin runner for tuple variable